### PR TITLE
Ensure atomic PID file writes

### PR DESCRIPTION
**Pull request by xantronix**
Ensure atomic PID file writes by writing PID to a temporary file named
for the PID, then renaming said file to the intended original filename

Other changes:

* Bump version to 0.96

### CODE REVIEW

File: lib/Unix/PID/Tiny.pm [hunk 1]
```diff
@@ -265,6 +267,8 @@ sub pid_file_no_unlink {
 
     syswrite( $pid_fh, int( abs($newpid) ) );
 
+    rename( $tmp_pid_file, $pid_file );
```
Comment (drmuey) [inline lib/Unix/PID/Tiny.pm:270]: should this be `rename(…) or return;` since it failing means we don’t have a pidfile in place?

Comment (drmuey) [reply lib/Unix/PID/Tiny.pm:270]: we should probably `or return;` for `_sysopen` and `syswrite` too now that I think about it.

Comment (drmuey) [reply lib/Unix/PID/Tiny.pm:270]: I put zero initially but I think `0` has a specific meaning in some places so a `return;` is a more general failure.

Comment (xantronix) [reply lib/Unix/PID/Tiny.pm:270]: I was just adhering to your specific lack of error handling in this case, but if you want some error handling then we can do that. :D How far do you think we should go?

Comment (drmuey) [reply lib/Unix/PID/Tiny.pm:270]: yeah I know, I think the Assume Success™ was an over sight. Maybe not too crazy, just on these file ops?

Comment (drmuey) [inline lib/Unix/PID/Tiny.pm:270]: oh yeah, almost forgot. `rename()` doesn’t work on open file handles on some systems I think we'd have to close it first, rename, then reopen it to get a new handle to make it more portable.

**Pull request closed without merging**